src/lib.rs

```diff
@@ -126,8 +139,81 @@ pub fn select_coin_knapsack(
 fn knap_sack(
     adjusted_target: u64,
     smaller_coins: &[(usize, OutputGroup)],
+    inputs: &[OutputGroup],
+    options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut target_reached = false;
```

Comment:
This variable is not needed, because if the target is reached,
best_set will not be empty and best_set_value will be less than u64::MAX